include/seqan3/alignment/pairwise/alignment_configurator.hpp

```diff
@@ -282,10 +286,42 @@ struct alignment_configurator
         if constexpr (config_t::template exists<align_cfg::band>())
             throw invalid_alignment_configuration{"Banded alignments are yet not supported."};
 
-        if constexpr (config_t::template exists<align_cfg::aligned_ends>())
-            throw invalid_alignment_configuration{"Free end gaps is not yet supported."};
+        // ----------------------------------------------------------------------------
+        // Configure semi-global alignment
+        // ----------------------------------------------------------------------------
+
+        // Get the value for the sequence ends configuration.
+        auto align_ends_cfg = cfg.template value_or<align_cfg::aligned_ends>(align_cfg::none_ends_free);
+        using align_ends_cfg_t = remove_cvref_t<decltype(align_ends_cfg)>;
+
+        auto configure_semi_global = [&](auto is_semi_global)
+        {
+            struct edit_traits_type
+            {
+                using word_type           [[maybe_unused]] = uint_fast64_t;
+                using is_semi_global_type [[maybe_unused]] = remove_cvref_t<decltype(is_semi_global)>;
+            };
 
-        return function_wrapper_t{edit_distance_wrapper<remove_cvref_t<config_t>>{cfg}};
+            return function_wrapper_t{edit_distance_wrapper<remove_cvref_t<config_t>, edit_traits_type>{cfg}};
+        };
+
+        // Check if it has free ends set for the first sequence trailing gaps.
+        auto has_free_ends_trailing = [&](auto first) constexpr
+        {
+            if constexpr (!decltype(first)::value)
+                return configure_semi_global(std::false_type{});
+            else if constexpr (align_ends_cfg_t::template is_static<1>())
+                return configure_semi_global(std::integral_constant<bool, align_ends_cfg_t::template get_static<1>()>{});
+            else // Resolve correct property at runtime.
+                return configure_semi_global(align_ends_cfg[1] ? std::true_type{} : std::false_type{});
+        };
+
+        // Check if it has free ends set for the first sequence leading gaps.
+        // If possible use static information.
+        if constexpr (align_ends_cfg_t::template is_static<0>())
+            return has_free_ends_trailing(std::integral_constant<bool, align_ends_cfg_t::template get_static<0>()>{});
+        else // Resolve correct property at runtime.
+            return has_free_ends_trailing(align_ends_cfg[0] ? std::true_type{} : std::false_type{});
```

Comment:
I doubt that this works :)

```suggestion
            if (align_ends_cfg[0])
                return has_free_ends_trailing(std::true_type{});
            else
                return has_free_ends_trailing(std::false_type{});
```

Reply:
Me too 🤣